File: solutions/beecrowd/1118/1118.cpp
#include <cstdint>
#include <cstdio>
#include <iostream>

int main() {
    std::int16_t r, c;
    float n, s;

    do {
        c = 0;
        s = 0.0;

        while (c < 2) {
            std::cin >> n;
            if (n >= 0.0 && n <= 10.0) {
                s += n;
                c++;
            } else {
                printf("nota invalida\n");
            }
        }

        printf("media = %.2f\n", s / 2.0);

        do {
            printf("novo calculo (1-sim 2-nao)\n");
            std::cin >> r;
        } while (r != 1 && r != 2);
    } while (r != 2);

    return 0;
}

File: solutions/beecrowd/1118/1118.py
while True:
    grade_sum = 0.0
    valid_grade_count = 0

    while valid_grade_count < 2:
        grade = float(input())

        if grade >= 0.0 and grade <= 10.0:
            grade_sum += grade
            valid_grade_count += 1
        else:
            print("nota invalida")

    print(f"media = {grade_sum / 2.0:.2f}")

    answer = input("novo calculo (1-sim 2-nao)\n")
    while answer not in ["1", "2"]:
        answer = input("novo calculo (1-sim 2-nao)\n")

    if answer == "2":
        break
